### PR TITLE
fix(scope): change execution block binding to be object that based a …

### PR DESCRIPTION
**Pull request by boc-tothefuture**
…block to rule

Addresses #224

### CODE REVIEW

**Review by jimtng — APPROVED** [review]
looks good

**Comment by jimtng** [comment]
Ooops forgot - please update the docs related to #{name}. These can be found in docs/examples/index.md and docs/usage/triggers/every.md

File: features/every.feature [hunk 1]
```diff
@@ -7,9 +7,9 @@ Feature:  every
   Scenario: Using cron shortcuts ':minute'
     Given a rule
       """
-      rule 'Log the rule name every minute' do
+      log_rule = rule 'Log the rule name every minute' do
```
Comment (pacive) [inline features/every.feature:10]: Could it be possible to pass the rule as a parameter to the rule block instead? Or would that not be available in the run-block's scope anyway?

```ruby
rule 'Log the rule name every minute' do |r|
  every :minute
  run { logger.info "Rule '#{r.name}' executed" }
end
```

**Review by pacive — COMMENTED** [review]
Does this change the behavior of other variables defined outside of rule blocks in any way? Perhaps even for the better?

File: features/every.feature (hunk 1 above)
Comment (boc-tothefuture) [reply features/every.feature:10]: That would in fact be better and much more ruby like..  right now after I look at this it wouldn't work with 'on start'.  Let me see if this works.

File: features/between.feature [hunk 2]
```diff
@@ -61,3 +61,21 @@ Feature: between
       | '<%=(Time.now - (5*60)).strftime('%H:%M:%S')%>'..'<%=(Time.now + (5*60)).strftime('%H:%M:%S')%>'  | '<%=(Time.now).strftime('%H:%M:%S')%>' | should     |
       | '<%=(Time.now + (5*60)).strftime('%H:%M:%S')%>'..'<%=(Time.now + (10*60)).strftime('%H:%M:%S')%>' | '<%=(Time.now).strftime('%H:%M:%S')%>' | should not |
 
+
+  Scenario Outline: Between ranges work inside of rule execution blocks
+    Given a rule template:
+      """
+      rule 'Testing Between Range' do
+        on_start
```
Comment (jimtng) [inline features/between.feature:69]: Do we need to also test this on other types of trigger, at least a cron trigger?

Comment (boc-tothefuture) [reply features/between.feature:69]: The every test is testing it with a cron trigger.

File: features/every.feature (hunk 1 above)
Comment (boc-tothefuture) [reply features/every.feature]: Updated..  
I am really glad we do code reviews.  this is much better.

File: features/between.feature (hunk 2 above)
Comment (jimtng) [reply features/between.feature:69]: I am referring to testing the between range. I'm not sure whether on_start runs in a different level of scope compared to the other triggers.

**Comment by boc-tothefuture** [comment]
> Ooops forgot - please update the docs related to #{name}. These can be found in docs/examples/index.md and docs/usage/triggers/every.md

Looks like you already took care of this in another PR?

**Comment by jimtng** [comment]
> > Ooops forgot - please update the docs related to #{name}. These can be found in docs/examples/index.md and docs/usage/triggers/every.md
> 
> Looks like you already took care of this in another PR?

I was just trying to be helpful and save you a few keystrokes. That PR is against the `run_context` branch (this PR). But obviously it can be discarded if we go with @pacive's suggestion which seems like a better way to go about it.

**Review by pacive — APPROVED** [review]
Looks good!

**Pull request merged**